src/ui.h

```diff
@@ -208,6 +211,8 @@ class uilist: public ui_container
         uilist( const std::string &hotkeys_override );
         // query() will be called at the end of these convenience constructors
         uilist( const std::string &msg, const std::vector<uilist_entry> &opts );
+        uilist( const std::string &msg, const std::vector<uilist_entry_pair_t> &opts,
+                const std::function<bool( uilist_entry_pair_t )> &f );
```

Comment:
This is a slightly weird interface.  You're allowing an arbitrary predicate to control which entries are included, but you're insisting that the only extra data that predicate has to work with is a `bool`.  I feel like it makes more sense to move this logic into `debug_menu.cpp`.  For example, something like:
```
std::vector<uilist_entry> filter_uilist(const std::vector<uilist_entry_pair_t> &, bool include_all)
...
     return uilist( _( "Player..." ), filter_uilist(uilist_initializer, display_all_entries) );
```

Reply:
Your absolutely right. Retrospectively I can see that I tried multiple things and ended up with a half-working solution to the problem.

I fixed it by using your suggestion.